### PR TITLE
LG-5389: progress bar QA improvements

### PR DESCRIPTION
**Pull request by charcoalyy**
## ✍️ Proposed changes

* use CSS variables instead of directly passing in variable to emotion, to prevent rebuilding styles on every change
* avoid coupling tests with implementation details
* explicitly guard against non-animated variants being passed into indeterminate bars

🎟 _Jira ticket:_ [LG-5389](https://jira.mongodb.org/browse/LG-5389)

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have run `pnpm changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: tools/install/src/ALL_PACKAGES.ts [hunk 1]
```diff
@@ -49,11 +50,13 @@ export const ALL_PACKAGES = [
   '@leafygreen-ui/polymorphic',
   '@leafygreen-ui/popover',
   '@leafygreen-ui/portal',
+  '@leafygreen-ui/preview-card',
   '@leafygreen-ui/progress-bar',
   '@leafygreen-ui/radio-box-group',
   '@leafygreen-ui/radio-group',
   '@leafygreen-ui/ripple',
   '@leafygreen-ui/search-input',
+  '@leafygreen-ui/section-nav',
```
Comment (TheSonOfThomp) [inline tools/install/src/ALL_PACKAGES.ts]: something you might be able to tackle once you're done here... ensure these get added in the correct "version packages" commit, not here

File: packages/progress-bar/src/ProgressBar/ProgressBar.tsx [hunk 2]
```diff
@@ -179,12 +179,19 @@ export function ProgressBar(props: ProgressBarProps) {
           >
             <div
               data-lgid={lgIds.fill}
+              style={
+                {
+                  // to avoid rebuilding styles on every width change, use CSS variables
```
Comment (TheSonOfThomp) [inline packages/progress-bar/src/ProgressBar/ProgressBar.tsx]: technically it's to avoid adding additional emotion style classes

Comment (TheSonOfThomp) [reply packages/progress-bar/src/ProgressBar/ProgressBar.tsx]: > use CSS variables

why? to do what?
(just a documentation nit)

File: packages/progress-bar/src/ProgressBar/ProgressBar.tsx [hunk 3]
```diff
@@ -179,12 +179,19 @@ export function ProgressBar(props: ProgressBarProps) {
...
+                  ...(!isIndeterminate && {
+                    '--width': `${displayWidth}%`,
+                    '--width-animation-duration': `${widthAnimationDuration}ms`,
+                  }),
+                } as React.CSSProperties
```
Comment (TheSonOfThomp) [inline packages/progress-bar/src/ProgressBar/ProgressBar.tsx]: nit: I feel like this might be cleaner if we defined the styles as a var outside the render block

File: packages/progress-bar/src/ProgressBar/ProgressBar.styles.ts [hunk 4]
```diff
@@ -166,18 +165,14 @@ const getDeterminateFillStyles = ({
...
   variant: Variant;
   disabled?: boolean;
-  width: number;
-  widthAnimationDuration?: number;
 }) => css`
-  width: ${width}%;
-  transition: width
-    ${widthAnimationDuration || DEFAULT_WIDTH_ANIMATION_DURATION}ms ease;
+  width: var(--width);
+  transition: var(--width-animation-duration) ease;
```
Comment (TheSonOfThomp) [inline packages/progress-bar/src/ProgressBar/ProgressBar.styles.ts:175]: Nit, since CSS vars are hard to trace, can we document here what element they're being defined on

File: packages/progress-bar/src/ProgressBar/ProgressBar.spec.tsx [hunk 5]
```diff
@@ -148,50 +140,6 @@ describe('packages/progress-bar', () => {
         expect(queryIcon()).toBeNull();
       });
     });
-
-    describe('with unexpected input', () => {
-      test('renders width 0% when value is less than 0', () => {
```
Comment (TheSonOfThomp) [inline packages/progress-bar/src/ProgressBar/ProgressBar.spec.tsx:153]: should we add some of these back as stories?

Comment (charcoalyy) [reply packages/progress-bar/src/ProgressBar/ProgressBar.spec.tsx:153]: since these edge cases are util-specific (i.e., resolved at props util, then propagated to rendering), i figured the util tests were sufficiently appropriate and avoids story clutter.
